drivers/adc/adc_ite_it8xxx2.c

```diff
@@ -0,0 +1,337 @@
+/*
+ * Copyright (c) 2020 ITE Corporation. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ite_it8xxx2_adc
+
+#define LOG_LEVEL CONFIG_ADC_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(adc_ite_it8xxx2);
+
+#include <drivers/adc.h>
+#include <soc.h>
+#include <errno.h>
+#include <assert.h>
+
+#define ADC_CONTEXT_USES_KERNEL_TIMER
+#include "adc_context.h"
+
+#define DEV_DATA(dev) \
+	((struct adc_it8xxx2_data * const)(dev)->data)
+
+/* External channels (maximum). */
+#define IT8XXX2_CHANNEL_COUNT 8
+
+/* Data structure to define ADC channel control registers. */
+struct adc_ctrl_t {
+	volatile uint8_t *adc_ctrl;
+	volatile uint8_t *adc_datm;
+	volatile uint8_t *adc_datl;
+};
+
+/* Data structure of ADC channel control registers. */
+const struct adc_ctrl_t adc_ctrl_regs[] = {
+	{&IT83XX_ADC_VCH0CTL, &IT83XX_ADC_VCH0DATM, &IT83XX_ADC_VCH0DATL},
+	{&IT83XX_ADC_VCH1CTL, &IT83XX_ADC_VCH1DATM, &IT83XX_ADC_VCH1DATL},
+	{&IT83XX_ADC_VCH2CTL, &IT83XX_ADC_VCH2DATM, &IT83XX_ADC_VCH2DATL},
+	{&IT83XX_ADC_VCH3CTL, &IT83XX_ADC_VCH3DATM, &IT83XX_ADC_VCH3DATL},
+	{&IT83XX_ADC_VCH4CTL, &IT83XX_ADC_VCH4DATM, &IT83XX_ADC_VCH4DATL},
+	{&IT83XX_ADC_VCH5CTL, &IT83XX_ADC_VCH5DATM, &IT83XX_ADC_VCH5DATL},
+	{&IT83XX_ADC_VCH6CTL, &IT83XX_ADC_VCH6DATM, &IT83XX_ADC_VCH6DATL},
+	{&IT83XX_ADC_VCH7CTL, &IT83XX_ADC_VCH7DATM, &IT83XX_ADC_VCH7DATL},
+};
+
+/* List of ADC channels. */
+enum chip_adc_channel {
+	CHIP_ADC_CH0 = 0,
+	CHIP_ADC_CH1,
+	CHIP_ADC_CH2,
+	CHIP_ADC_CH3,
+	CHIP_ADC_CH4,
+	CHIP_ADC_CH5,
+	CHIP_ADC_CH6,
+	CHIP_ADC_CH7,
+	CHIP_ADC_COUNT,
+};
+
+struct adc_it8xxx2_data {
+	struct adc_context ctx;
+	uint16_t *buffer;
+	uint16_t *repeat_buffer;
+	uint8_t adc_ch;
+	uint8_t resolution;
+};
+static void adc_it8xxx2_isr(const void *arg);
+
+static int adc_it8xxx2_channel_setup(const struct device *dev,
+				 const struct adc_channel_cfg *channel_cfg)
+{
+	struct adc_it8xxx2_data *data = DEV_DATA(dev);
+
+	data->adc_ch = channel_cfg->channel_id;
+
+	if (channel_cfg->acquisition_time != ADC_ACQ_TIME_DEFAULT) {
+		return -EINVAL;
+	}
+
+	if (data->adc_ch >= IT8XXX2_CHANNEL_COUNT) {
```

Comment:
```suggestion
	if (data->adc_ch > IT8XXX2_CHANNEL_COUNT) {
```

Reply:
If the channel is 8, it is invalid, so equal is needed.

Reply:
Of course it is invalid. My bad. I don't know actually what I was thinking about when adding this suggestion. Sorry for that.